File: Engine/src/jvmMain/java/org/etieskrill/engine/graphics/gl/shader/impl/SingleColourShader.java
package org.etieskrill.engine.graphics.gl.shader.impl;

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader;
import org.etieskrill.engine.graphics.shader.Shader;

import java.util.List;

import static org.etieskrill.engine.graphics.shader.Shader.Uniform.Type.VEC4;

@ReflectShader
public class SingleColourShader extends Shader {
    public SingleColourShader() {
        super(List.of("SingleColour.vert", "SingleColour.frag"), List.of(uniform("colour", VEC4)));
    }
}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/graphics/shader/impl/SingleColourShader.kt
package org.etieskrill.engine.graphics.shader.impl

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader
import org.etieskrill.engine.graphics.shader.Shader
import org.joml.Matrix4f
import org.joml.Vector4f

@ReflectShader
class SingleColourShader : Shader(listOf("shaders/SingleColour.vert", "shaders/SingleColour.frag")) {
    init {
        addUniform("model", Matrix4f::class)
        addUniform("combined", Matrix4f::class)
        addUniform("colour", Vector4f::class)
    }
}
